sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaRulesDefinition.java

```diff
@@ -151,10 +160,16 @@ private static void addMetadata(NewRule rule, @Nullable RuleMetadata metadata) {
     addSecurityStandards(rule, metadata.securityStandards);
   }
 
-  private static void addSecurityStandards(NewRule rule, SecurityStandards securityStandards) {
-    for (String s : securityStandards.OWASP) {
+  private void addSecurityStandards(NewRule rule, SecurityStandards securityStandards) {
+    for (String s : securityStandards.OWASP_2017) {
       rule.addOwaspTop10(OwaspTop10.valueOf(s));
     }
+    boolean isOwaspByVersionSupported = sonarRuntime.getApiVersion().isGreaterThanOrEqual(Version.create(9, 3));
+    if (isOwaspByVersionSupported) {
+      for (String s : securityStandards.OWASP_2021) {
+        rule.addOwaspTop10(OwaspTop10Version.Y2021, OwaspTop10.valueOf(s));
```

Comment:
I think I would prefer to have the enums `OwaspTop10Version` and `OwaspTop10` prefixed with `RulesDefinition` to be clear about where they come from:

```
RulesDefinition.OwaspTop10Version
```